File: src/main/java/employeetech/com/employee/manager/service/StudentService.java
package employeetech.com.employee.manager.service;


import employeetech.com.employee.manager.model.Student;
import employeetech.com.employee.manager.repo.StudentRepo;
import jakarta.transaction.Transactional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
@Transactional
public class StudentService {
      public  final StudentRepo studentRepo;

      @Autowired
      public  StudentService(StudentRepo studentRepo) { this.studentRepo = studentRepo;}

    public Student addStudent( Student student ){
          return studentRepo.save(student);
    }

     public Student getStudentById(Long id){
          return  studentRepo.findById(id)
                  .orElseThrow(() -> new RuntimeException( "Student is not found"));
     }

     public Student updateStudent(Student student ){
          return  studentRepo.save(student);
     }
}

File: src/main/kotlin/employeetech/com/employee/manager/service/StudentService.kt
package employeetech.com.employee.manager.service

import employeetech.com.employee.manager.model.Student
import employeetech.com.employee.manager.repo.StudentRepo
import jakarta.transaction.Transactional
import org.springframework.stereotype.Service

@Service
@Transactional
class StudentService(private val studentRepo: StudentRepo) {

    fun addStudent(student: Student): Student {
        return studentRepo.save(student)
    }

    fun findAllStudents(): List<Student> {
        return studentRepo.findAll()
    }

    fun getStudentById(id: Long): Student {
        return studentRepo.findById(id)
            .orElseThrow { RuntimeException("Student was not found") }
    }

    fun updateStudent(student: Student): Student {
        return studentRepo.save(student)
    }

    fun deleteStudent(id: Long) {
        studentRepo.deleteById(id)
    }
}
